ml-agents/mlagents/plugins/stats_writer.py

```diff
@@ -0,0 +1,62 @@
+import sys
+from typing import List
+
+# importlib.metadata is new in python3.8
+# We use the backport for older python versions.
+if sys.version_info < (3, 8):
+    import importlib_metadata
+else:
+    import importlib.metadata as importlib_metadata  # pylint: disable=E0611
+
+from mlagents.trainers.stats import StatsWriter
+
+from mlagents_envs import logging_util
+from mlagents.trainers.settings import RunOptions
+from mlagents.trainers.stats import TensorboardWriter, GaugeWriter, ConsoleWriter
+
+
+logger = logging_util.get_logger(__name__)
+
+
+def get_default_stats_writers(run_options: RunOptions) -> List[StatsWriter]:
+    """
+    The StatsWriters that mlagents-learn always uses:
+    * A TensorboardWriter to write information to TensorBoard
+    * A GaugeWriter to record our internal stats
+    * A ConsoleWriter to output to stdout.
+    """
+    checkpoint_settings = run_options.checkpoint_settings
+    return [
+        TensorboardWriter(
+            checkpoint_settings.write_path,
+            clear_past_data=not checkpoint_settings.resume,
+        ),
+        GaugeWriter(),
+        ConsoleWriter(),
+    ]
+
+
+def register_stats_writer_plugins(run_options: RunOptions) -> List[StatsWriter]:
+    """
+    Registers all StatsWriter plugins (including the default one),
+    and evaluates them, and returns the list of all the StatsWriter implementations.
+    """
+    all_stats_writers: List[StatsWriter] = []
+    entry_points = importlib_metadata.entry_points()["mlagents.stats_writer"]
+
+    for entry_point in entry_points:
+
+        try:
+            logger.debug(f"Initializing StatsWriter plugins: {entry_point.name}")
+            plugin_func = entry_point.load()
+            plugin_stats_writers = plugin_func(run_options)
+            logger.debug(
+                f"Found {len(plugin_stats_writers)} StatsWriters for plugin {entry_point.name}"
+            )
+            all_stats_writers += plugin_stats_writers
+        except BaseException:
```

Comment:
I think catching the error is great, but the error should be displayed as well / be logged somewhere.

Reply:
`logging.exception` will print the exception stack trace and information:
``` python
>>> import logging
>>> logger = logging.getLogger(__name__)
>>> try:
...   int('foo')
... except:
...   logger.exception("oops")
... 
oops
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: invalid literal for int() with base 10: 'foo'
```